argo/kfp-compiler/Makefile

```diff
@@ -13,22 +13,27 @@ penguin-staging:
 	cp ../../docs-gen/includes/quickstart/penguin_pipeline/*.py $(PENGUIN_STAGING)
 	cp integration/Dockerfile $(PENGUIN_STAGING)
 
-penguin-37: penguin-staging
-	docker build --build-arg PYTHON_VERSION=3.7.12 --build-arg TFX_VERSION=1.9.1 -t penguin:3.7 $(PENGUIN_STAGING)
+penguin-310: penguin-staging
+	docker build --build-arg PYTHON_VERSION=3.10.12 --build-arg TFX_VERSION=1.14.0 -t penguin:3.10 $(PENGUIN_STAGING)
 
 penguin-39: penguin-staging
 	docker build --build-arg PYTHON_VERSION=3.9.13 --build-arg TFX_VERSION=1.9.1 -t penguin:3.9 $(PENGUIN_STAGING)
 
-integration-test: docker-build penguin-37 penguin-39
+integration-test: docker-build penguin-310 penguin-39
 	$(eval TMP := $(shell mktemp -d))
+	echo '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}' > $(TMP)/pipeline_config.json
 	docker run -v $(TMP):/shared kfp-operator-argo-kfp-compiler /shared
-	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.7 --output_file /tmp/pipeline.yaml --pipeline_config '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}'
-	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config '{"rootLocation": "pipeline_root", "servingLocation": "serving", "name": "test", "image": "test-pipeline", "tfxComponents": "pipeline.create_components"}'
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json --execution_mode v1
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.10 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json  --execution_mode v1
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.9 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json --execution_mode v2
+	docker run -v $(TMP):/shared --entrypoint /shared/compile.sh penguin:3.10 --output_file /tmp/pipeline.yaml --pipeline_config /shared/pipeline_config.json  --execution_mode v2
 
 ##@ Build
 
 build:
+	poetry env use /home/jenkins/.asdf/installs/python/3.10.12/bin/python
 	pip install poetry-dynamic-versioning --quiet
+	poetry config virtualenvs.create false
```

Comment:
This doesn't seem to be either necessary or a good idea. It changes the config for the user (in my case it changed `~/Library/Preferences/pypoetry/config.toml` that is never what we want). Also we don't want to install anything in the system packages:
> If set to false, Poetry will not create a new virtual environment. If it detects an already enabled virtual environment or an existing one in `{cache-dir}/virtualenvs` or `{project-dir}/.venv` it will install dependencies into them, otherwise it will install dependencies into the systems python environment.

Reply:
fixed